x-pack/legacy/plugins/lens/public/redirect.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// This file redirects lens urls starting with app/lens#... to their counterpart on app/kibana#lens/... to
+// make sure it's compatible with the 7.5 release
+
+const newUrl =
+  // replace app/lens in the path with app/kibana
+  window.location.pathname.replace(/app\/lens$/, 'app/kibana') +
+  // prefix the path in the url with lens
+  window.location.hash.replace(/^#\//, '#/lens/');
+
+// redirect
+window.location.href = window.location.href.replace(
```

Comment:
This code seems a bit brittle. Can we do something to avoid replacing the location in such a way?

Reply:
This code is covering a very specific edge case - when a user bookmarks a Lens visualization in 7.5

As it's just here for BWC reasons I wanted to keep it as simple as possible. Do you have a suggestion for improvement?

Reply:
I agree with Maja. It'll still be hacky, but I'd be more OK with this if it was done inside a conditional, rather than 100% of the time that Lens loads up. (e.g. if we detect we're on a 7.5 URl, redirect to a 7.6 URL, etc).

Reply:
Ah, there might be a misconception. This isn't executed each time Lens loads - just if you navigate to `app/lens` (which can only happen if you have a bookmark stored because the Visualize dialog will point to `app/kibana#/lens` instead).